packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -405,19 +405,20 @@ export class DatabaseCluster extends DatabaseClusterBase {
 
     const clusterAssociatedRoles: CfnDBCluster.DBClusterRoleProperty[] = [];
     let { s3ImportRole, s3ExportRole } = this.setupS3ImportExport(props);
-    if (s3ImportRole) {
-      clusterAssociatedRoles.push({ roleArn: s3ImportRole.roleArn });
-    }
-    if (s3ExportRole) {
-      clusterAssociatedRoles.push({ roleArn: s3ExportRole.roleArn });
-    }
-
     // bind the engine to the Cluster
     const clusterEngineBindConfig = props.engine.bindToCluster(this, {
       s3ImportRole,
       s3ExportRole,
       parameterGroup: props.parameterGroup,
     });
+
+    if (s3ImportRole) {
```

Comment:
Can you move `clusterAssociatedRoles` here as well? In general, variables should be defined as close to where they are used as possible.